cupyx/scipy/sparse/base.py

```diff
@@ -434,9 +434,28 @@ def multiply(self, other):
     def power(self, n, dtype=None):
         return self.tocsr().power(n, dtype=dtype)
 
-    def reshape(self, shape, order='C'):
-        """Gives a new shape to a sparse matrix without changing its data."""
-        raise NotImplementedError
+    def reshape(self, *args, order='C'):
```

Comment:
Would you change the parameter name to `shape` to get along with `cupy.ndarray.reshape`? I know SciPy has the name `args` and overwrites it by docstring on the documentation. It is an exceptional case.
```suggestion
    def reshape(self, *shape, order='C'):
```